TM1py/Services/ElementService.py

```diff
@@ -168,11 +168,65 @@ def get_element_identifiers(self, dimension_name: str, hierarchy_name: str,
              """.format(
             elem_mdx=mdx_element_selection,
             attr_mdx=",".join(build_element_unique_names(
-                ["}ElementAttributes_" + dimension_name] * len(alias_attributes),
-                alias_attributes)),
+                ["}ElementAttributes_" + dimension_name] * len(alias_attributes), alias_attributes)),
             dim=dimension_name)
         return self._retrieve_mdx_rows_and_cell_values_as_string_set(mdx, **kwargs)
 
+    def get_element_by_attribute(self, dimension_name: str, hierarchy_name: str, attribute: str,
+                                 elements: Union[str, List[str]] = None, **kwargs) -> dict:
+        """
+         Get element name and attribute value for a set of elements in a hierarchy
+
+        :param dimension_name:
+        :param hierarchy_name:
+        :param attribute: Name of the Attribute
+        :param elements:  MDX (Set) expression or iterable of elements
+        :keyword exclude_empty_cells: Boolean
+        :keyword element_unique_names: Boolean
```

Comment:
Why use `kwargs` for these two arguments?

I would prefer to have them as optional arguments with default values. Then PyCharm, and other smart IDEs, would recognize them as arguments to the function.
Currently, we use `kwargs` almost exclusively for "technical" arguments that are consistent throughout the entire library, such as, `timeout`

Reply:
yeah makes more sense to make them Default.